### PR TITLE
Two year secondary pow ratio fix

### LINKED ISSUE #1978: fix secondary_pow_ratio to take 2 years rather than 90 weeks (1.73 years)
90u64.saturating_sub(height / WEEK_HEIGHT)

should read

	90u64.saturating_sub(height  / (2*YEAR_HEIGHT / 90))

Note that DAY_HEIGHT and hence YEAR_HEIGHT are in fact divisible by 90.

### PR DESCRIPTION
**Pull request by antiochp**
Resolves #1978.

Locks the change in to activate as part of the scheduled `T4_CUCKAROO_HARDFORK (height: 64_000)` hard fork.

Q) Do we want to do this as part of this hard fork?

### CODE REVIEW

File: core/src/consensus.rs [hunk 1]
```diff
@@ -63,7 +63,11 @@ pub const COINBASE_MATURITY: u64 = DAY_HEIGHT;
 /// function of block height (time). Starts at 90% losing a percent
 /// approximately every week. Represented as an integer between 0 and 100.
 pub fn secondary_pow_ratio(height: u64) -> u64 {
```
Comment (ignopeverell) [inline core/src/consensus.rs]: That means we would start mainnet with the same slightly different schedule? Should we check the global chain type as well here?

**Comment by ignopeverell** [comment]
Doing this with the HF is a good idea. But I think we should start with the right calculation right away on mainnet.

**Comment by antiochp** [comment]
I was assuming we would be cleaning up the T4 hardfork code prior to mainnet.
But yeah - let me add the global chaintype here also just to be sure.

Agreed we should start with the correct calc from the first block on mainnet.

**Comment by ignopeverell** [comment]
I don't think we'll be able to do cleanup with T4, at least as long as we keep it around.

**Comment by tromp** [comment]
I don't think we should keep T4 around post launch. We should just clean up the code and launch the real testnet shortly before mainnet, so they run exactly the same code (but different genesis).

**Comment by ignopeverell** [comment]
I guess that's something we need to discuss then.

**Comment by antiochp** [comment]
Ah of course - we want _some_ testnet running in parallel with maninet, and I was forgetting about that.

:+1:

**Comment by antiochp** [comment]
We activate this at `T4_CUCKAROO_HARDFORK` which is in a couple of days. Is this long enough to reasonably expect the network to upgrade by then?

@ignopeverell @tromp

**Comment by tromp** [comment]
It's a little too soon, given that https://github.com/mimblewimble/grin/issues/1980 is not part of the HF yet. I would suggest delaying it by about a week. By that time we might have a cuckaroo cpu miner ready as well...

**Comment by antiochp** [comment]
Hmm ok. I'll add a new hardfork height and activate this (and presumably #1980) at that point.

We'll need to work out what do do with anybody who deployed between this PR getting merged and the new hardfork height being introduced...

**Comment by tromp** [comment]
Right; people who don't update to the new HF height in time will find themselves on a minority fork...

**Comment by ignopeverell** [comment]
From Gitter:

> I wouldn't mind merging mimblewimble/grin#2050 now for the HF tomorrow, creating a little mayhem on T4 wouldn't be a bad thing, and make sure things behave appropriately in the few days after that

**Pull request merged**